pylint/checkers/classes.py

```diff
@@ -867,14 +869,16 @@ def _check_proper_bases(self, node):
 
     def _check_typing_final(self, node: nodes.ClassDef) -> None:
         """Detect that a class does not subclass a class decorated with `typing.final`"""
-        if not PY38_PLUS:
+        if not self._py38_plus:
             return
         for base in node.bases:
             ancestor = safe_infer(base)
             if not ancestor:
                 continue
-            if isinstance(ancestor, nodes.ClassDef) and decorated_with(
-                ancestor, ["typing.final"]
+
+            if isinstance(ancestor, nodes.ClassDef) and (
+                decorated_with(ancestor, ["typing.final"])
+                or (uninferable_final_decorators(ancestor.decorators))
```

Comment:
The outer bracket here is not necessary. I've opened #5300 to remove them.

Reply:
Super thanks for doing this @cdce8p!